pennylane/ops/functions/equal.py

```diff
@@ -269,12 +269,12 @@ def _equal_operators(
 # pylint: disable=unused-argument, protected-access
 def _equal_prod_and_sum(op1: CompositeOp, op2: CompositeOp, **kwargs):
     """Determine whether two Prod or Sum objects are equal"""
-    if len(op1.operands) != len(op2.operands):
```

Comment:
Since this change impacts `Prod` as well could you also add a test for that? I.e comparing a prod of prods with a prod.

Reply:
Good call 👍 ✅